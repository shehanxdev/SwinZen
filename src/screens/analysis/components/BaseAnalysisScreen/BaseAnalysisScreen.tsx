import React from 'react';
import { SafeAreaView, ScrollView } from 'react-native';

import { tw } from '@sz/config';
import { useHeaderHeight } from '@sz/hooks';

//This component is needed since, Analysis screen has a different base screen
export function BaseAnalysisScreen({ children }) {
  const headerHeight = useHeaderHeight();

  return (
    <SafeAreaView style={tw`h-full pt-[${headerHeight}px] bg-Neutral-Sz900`}>
      <ScrollView contentContainerStyle={tw`grow`}>{children}</ScrollView>
    </SafeAreaView>
  );
}
