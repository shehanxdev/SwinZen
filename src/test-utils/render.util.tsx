import { NavigationContext } from '@react-navigation/native';
import { render } from '@testing-library/react-native';
import React, { ReactNode } from 'react';
import Config from 'react-native-config';
import { Provider as PaperProvider } from 'react-native-paper';
import { Provider } from 'react-redux';

import { paperTheme } from '@sz/config';
import { ConfigService } from '@sz/services';
import { store } from '@sz/stores';

type RenderParams = Parameters<typeof render>;

export function renderWithProviders(ui: RenderParams[0], options?: RenderParams[1]) {
  ConfigService.setCallback(key => Config[key]);

  const BaseProviders = ({ children }: { children: ReactNode }) => (
    <Provider store={store}>
      <NavigationContext.Provider
        //@ts-ignore
        value={{
          isFocused: () => true,
          addListener: jest.fn(() => jest.fn()),
        }}>
        <PaperProvider theme={paperTheme}>{children}</PaperProvider>
      </NavigationContext.Provider>
    </Provider>
  );

  return render(ui, { wrapper: BaseProviders, ...options });
}
