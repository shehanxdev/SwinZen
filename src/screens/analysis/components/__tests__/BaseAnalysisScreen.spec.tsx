import * as React from 'react';

import { renderWithProviders } from '@sz/test-utils';

import { BaseAnalysisScreen } from '../BaseAnalysisScreen';

describe('BaseAnalysis Screen', () => {
  const testID = 'BaseAnalysisScreenTestID';
  const getRenderedScreen = () =>
    renderWithProviders(
      <BaseAnalysisScreen>
        <></>
      </BaseAnalysisScreen>,
    );

  it(`should render BaseAnalysis Screen correctly`, async () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the BaseAnalysis Screen via testID', () => {
    const { getByTestId } = getRenderedScreen();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeDefined();
  });
});
