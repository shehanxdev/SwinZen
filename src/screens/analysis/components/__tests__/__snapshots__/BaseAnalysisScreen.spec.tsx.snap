// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseAnalysis Screen should render BaseAnalysis Screen correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      {
        "backgroundColor": "#212529",
        "height": "100%",
        "paddingTop": 50,
      }
    }
    testID="BaseAnalysisScreenTestID"
  >
    <RCTScrollView
      contentContainerStyle={
        {
          "flexGrow": 1,
        }
      }
    >
      <View />
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
